### PR TITLE
Creating input to customize the duration for a pipeline

### PR DESCRIPTION
**Pull request by gdelgadot**
Signed-off-by: Gloria Delgado <gloriad@vmware.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -49,6 +49,7 @@ The above line is using the GitHub Action default input parameters. You can cust
 | retry-count            | This is the default number of retries to do in case of failure reaching out to VIB.                                                                 | 3                    |
 | upload-artifacts       | This parameter specifies whether the GitHub Action will publish logs and reports as GitHub artifacts.                                               | true                 |
 | verification-mode      | This parameter changes the default parallel verification mode to serial.                                                                   | PARALLEL             |
+| pipeline-duration      | This parameter specifies the time for a pipeline execution to be completed.                                                                  | 90 * 60 * 1000              |
```
Comment (mpermar) [inline README.md:52]: Lets change this name to max-pipeline-duration

File: action.yml [hunk 2]
```diff
@@ -37,6 +37,10 @@ inputs:
     description: 'When specified, this GitHub Action will request VIB to execute the pipeline in the requested verification mode. Possible values: SERIAL, PARALLEL.'
     required: false
     default: PARALLEL
+  pipeline-duration:
+    description: 'Time for a pipeline execution to be completed.'
```
Comment (mpermar) [inline action.yml:41]:
```suggestion
    description: 'Maximum time for a pipeline execution to be completed. The value should be in seconds.' 
```

File: src/constants.ts [hunk 3]
```diff
@@ -15,9 +15,9 @@ export const DEFAULT_PIPELINE = "vib-pipeline.json"
 /**
  * Max waiting time for an execution graph to complete
  *
- * @default 90 minutes
+ * @default 6 hours
```
Comment (mpermar) [inline src/constants.ts:18]: 6h is too much. I think we can have our default set to 2h but at the same time you need another global variable that sets the max time that GitHub will let your run an action on. That can be MAX_GITHUB_ACTION_RUN_TIME and should be set to 6 hours.

**Review by mpermar — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (gdelgadot) [reply README.md:52]: done

File: action.yml [hunk 4]
```diff
@@ -37,6 +37,10 @@ inputs:
     description: 'When specified, this GitHub Action will request VIB to execute the pipeline in the requested verification mode. Possible values: SERIAL, PARALLEL.'
     required: false
     default: PARALLEL
+  pipeline-duration:
+    description: 'Time for a pipeline execution to be completed.' 
+    required: false
+    default: 90 * 60 * 1000
```
[the remark this answers is not in the document]
Comment (gdelgadot) [reply action.yml:43]: done

File: src/constants.ts (hunk 3 above)
Comment (gdelgadot) [reply src/constants.ts:18]: done

File: src/main.ts [hunk 5]
```diff
@@ -105,12 +105,18 @@ export async function runAction(): Promise<any> {
...
     let executionGraph = await getExecutionGraph(executionGraphId)
+    const pipelineDuration = getNumberInput("pipeline-duration")
     while (!Object.values(constants.EndStates).includes(executionGraph["status"])) {
       core.info(`  » Pipeline is still in progress, will check again in 15s.`)
-      if (Date.now() - startTime > constants.DEFAULT_EXECUTION_GRAPH_GLOBAL_TIMEOUT) {
+      if (Date.now() - startTime > pipelineDuration) {
         //TODO: Allow user to override the global timeout via action input params
         core.info(`Pipeline ${executionGraphId} timed out. Ending Github Action.`)
         break
+      } else if (pipelineDuration > constants.DEFAULT_EXECUTION_GRAPH_GLOBAL_TIMEOUT) {
```
[the remark this answers is not in the document]
Comment (gdelgadot) [reply src/main.ts:115]: done

File: action.yml (hunk 2 above)
Comment (gdelgadot) [reply action.yml]: done

File: src/constants.ts [hunk 6]
```diff
@@ -15,9 +15,16 @@ export const DEFAULT_PIPELINE = "vib-pipeline.json"
 /**
  * Max waiting time for an execution graph to complete
  *
- * @default 90 minutes
+ * @default 2 hours
```
Comment (mpermar) [inline src/constants.ts:18]: This value does not match the documentation. Lets change it back to 90mins. That's ok.

File: README.md [hunk 7]
```diff
@@ -49,6 +49,7 @@ The above line is using the GitHub Action default input parameters. You can cust
 | retry-count            | This is the default number of retries to do in case of failure reaching out to VIB.                                                                 | 3                    |
 | upload-artifacts       | This parameter specifies whether the GitHub Action will publish logs and reports as GitHub artifacts.                                               | true                 |
 | verification-mode      | This parameter changes the default parallel verification mode to serial.                                                                   | PARALLEL             |
+| max-pipeline-duration      | This parameter specifies the time for a pipeline execution to be completed.                                                                  | 5400              |
```
Comment (mpermar) [inline README.md:52]:
```suggestion
| max-pipeline-duration      | This parameter specifies the time in seconds for a pipeline execution to be completed.                                                                  | 5400              |
```

File: dist/index.js [hunk 8]
```diff
@@ -310,10 +316,14 @@ function runAction() {
             core.info(`Starting the execution of the pipeline with id ${executionGraphId}, check the pipeline details: ${getDownloadVibPublicUrl()}/v1/execution-graphs/${executionGraphId}`);
             // Now wait until pipeline ends or times out
             let executionGraph = yield getExecutionGraph(executionGraphId);
+            let pipelineDuration = getNumberInput("max-pipeline-duration");
```
Comment (mpermar) [inline dist/index.js:319]: Just realized we are missing a `* 1000` here as the input value will be in seconds.

**Review by mpermar — CHANGES_REQUESTED** [review]

File: README.md (hunk 7 above)
Comment (gdelgadot) [reply README.md:52]: done

File: dist/index.js (hunk 8 above)
Comment (gdelgadot) [reply dist/index.js:319]: done

**Review by mpermar — APPROVED** [review]

**Pull request merged**